index.d.ts

```diff
@@ -0,0 +1,169 @@
+  hash: string;
+  input: string;
+  nonce: string;
+  to: string;
+  value: string;
+  v: string;
+  r: string;
+  s: string;
+  privateFrom: string;
+  privateFor?: string[];
+  privacyGroupId?: string;
+  restriction: string;
+};
+
+export default class EEAClient extends Web3 {
+  constructor(web3: Web3, chainId?: number);
+  eea: {
+    sendRawTransaction: (options: SendRawOptions) => Promise<string>;
+  };
+  priv: {
+    generatePrivacyGroup: (options: { privateFrom: string; privateFor?: string | string[] }) => string;
+    deletePrivacyGroup: (options: { privacyGroupId: string }) => Promise<string>;
+    findPrivacyGroup: (options: { addresses: string[] }) => Promise<PrivacyGroup[]>;
+    distributeRawTransaction: (options: SendRawOptions) => Promise<string>;
+    getTransactionCount: (options: { from: string; privacyGroupId?: string }) => Promise<number>;
+    getTransactionReceipt: (
+      txHash: string,
+      enclavePublicKey?: string,
+      retries?: number,
+      delay?: number
+    ) => Promise<PrivateTransactionReceipt | null>;
+    call: (options: { privacyGroupId: string; to: string; data: string; blockNumber?: number }) => Promise<PrivateTransactionReceipt>;
+    subscribe: (
+      privacyGroupId: string,
+      filter: Filter,
+      callback: (error: Error | null, result?: string) => void
+    ) => Promise<PrivateSubscription>;
+    createPrivacyGroup: (options: { addresses: string[]; name?: string; description?: string }) => Promise<string>;
+    getTransaction: (txHash: string) => Promise<PrivateTransaction | null>;
+    getPastLogs: (privacyGroupId: string, filter: Filter) => Promise<Log[]>;
+    createFilter: (privacyGroupId: string, filter: Filter) => Promise<string>;
+    getFilterLogs: (privacyGroupId: string, filterId: string) => Promise<Log[]>;
+    getFilterChanges: (privacyGroupId: string, filterId: string) => Promise<Log[]>;
+    uninstallFilter: (privacyGroupId: string, filterId: string) => Promise<boolean>;
+  };
+  privx: {
+    createPrivacyGroup: (
+      options: {
+        privacyGroupId?: string;
+        privateKey: string;
+        enclaveKey: string;
+        participants: string[];
+      }
+    ) => Promise<PrivateTransactionReceipt>;
+    findOnChainPrivacyGroup: (options: { addresses: string[] }) => Promise<PrivacyGroup>;
+    removeFromPrivacyGroup: (
+      options: {
+        privacyGroupId: string;
+        privateKey: string;
+        enclaveKey: string;
+        participant: string;
+      }
+    ) => Promise<PrivateTransactionReceipt>;
+    addToPrivacyGroup: (
+      options: {
+        privacyGroupId: string;
+        privateKey: string;
+        enclaveKey: string;
+        participants: string[];
+      }
+    ) => Promise<PrivateTransactionReceipt>;
+    setPrivacyGroupLockState: (
+      options: {
+        privacyGroupId: string;
+        privateKey: string;
+        enclaveKey: string;
+        lock: boolean;
+      }
+    ) => Promise<PrivateTransactionReceipt>;
+  };
+}
+
```

Comment:
looks pretty good so far. This should help readability. Thanks for your contribution @epiccoolguy